components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -159,6 +161,39 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
             return Task.CompletedTask;
         }
 
+        /// <inheritdoc/>
+        public Task<JobDiagnosticInfoModel> GetDiagnosticInfo() {
```

Comment:
Is there anything stopping us from calling `GetDiagnosticInfo()` from within `DiagnosticsOutputTimer_Elapsed()` method ? That way we can keep all the computation of the values in `GetDiagnosticInfo()` (instead of duplicating it in both places) and just consume the calculated result in `DiagnosticsOutputTimer_Elapsed()`.

Reply:
no, now called in DiagnosticsOutputTimer_Elapsed